### ISSUE TITLE
ignore: test paths for ignore::Match

### ISSUE THREAD

**Issue by dns2utf8** [BurntSushi/ripgrep#763]
I would like to watch all the files of a source tree.
Currently I setup the file watches for each file individually and reset the whole thing if I notice a new folder.

Is it currently possible to match a Path against a stored object? My guess is, probably not the walker?

Best,
Stefan

**Comment by BurntSushi** [comment]
There is nothing out of the box in the `ignore` crate that will do this for you, no. But the building blocks are there.

**Comment by dns2utf8** [comment]
So can I build a matcher outside of the ignore crate or would I have to extend it?

**Comment by BurntSushi** [comment]
I don't understand your question.

**Comment by dns2utf8** [comment]
Let me try again: In your opinion, would you try to implement this feature inside the `ignore` crate? Or can I do it with the public interface?

**Comment by BurntSushi** [comment]
@dns2utf8 I think there is a place for it in the `ignore` crate, but it should be doable using the existing public API of `ignore`. That is, you shouldn't need any internals to build it. Note that this is not an easy thing to get right!

**Comment by BurntSushi** [comment]
I am going to close this issue because I personally don't have any plans to work on this, and I'm not sure I even have the bandwidth to mentor it. @dns2utf8 If you want to ask more questions though, then doing it in this issue is fine and I'd be open to re-opening it if someone really wanted to take this on.
